### DISCUSSION TITLE
@parcel/source-map fails to install on iOS

### DISCUSSION THREAD

**Discussion by danieltroger** [parcel-bundler/parcel#6190] [Q&A] [answered]
Hi,
I'm trying to install parcel 2 on my iPhone because I imagine saying "I built \<x> on my iPhone" sounds very cool.

I'm running an iPhone X on iOS 14.3 with node version `v14.13.1` and npm version `6.14.8` and it seems to work way better than on my iPad with node version 12 where v8 ran out of heap space or something.

BUT: @parcel/source-map does not install and it's hard to see from where the error comes.

I remember reading somewhere the source-map thing is written in c++, maybe that is the issue? Do I have to install clang or something so that it compiles? How does it work?

Here's the log file:
[2021-04-26T22_00_39_367Z-debug.log](https://github.com/parcel-bundler/parcel/files/6380620/2021-04-26T22_00_39_367Z-debug.log)

And here's the output of `npm install` (the hostname of my iPhone is `Desktop-cf97a`):

```
Desktop-cf97a:/tmp/parcel-doesnt-declare-its-variables mobile% npm install
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported

> @parcel/source-map@2.0.0-alpha.4.21 install /private/var/tmp/parcel-doesnt-declare-its-variables/node_modules/@parcel/source-map
> node-gyp-build

/private/var/tmp/parcel-doesnt-declare-its-variables/node_modules/.bin/node-gyp-build: 3: Syntax error: "(" unexpected
npm WARN parcel-doesnt-declare-its-variables No description
npm WARN parcel-doesnt-declare-its-variables No repository field.
npm WARN parcel-doesnt-declare-its-variables No license field.

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @parcel/source-map@2.0.0-alpha.4.21 install: `node-gyp-build`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the @parcel/source-map@2.0.0-alpha.4.21 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /var/mobile/.npm/_logs/2021-04-26T22_00_39_367Z-debug.log
```

` Syntax error: "(" unexpected` sounds fixable but I can't find any further information from where it comes. Would appreciate some help 😁

**Comment by mischnic** [comment, marked as answer]
What happens if you 
```sh
$ cd /private/var/tmp/parcel-doesnt-declare-its-variables/node_modules/@parcel/source-map
```
and then run
```sh
$ npx node-gyp-build
# or
$ /private/var/tmp/parcel-doesnt-declare-its-variables/node_modules/.bin/node-gyp-build
```
directly?

**Reply by danieltroger** [reply]
Solution - for this problem - was to make sure node is running node-gyp-build and not bash/zsh. Accepting this answer since I can't accept the comment.

**Comment by danieltroger** [comment]
> What happens if you

it seems like npm unfortuately removes it as cleanup:

```
Desktop-cf97a:/private/var/tmp/parcel-doesnt-declare-its-variables/node_modules/@parcel mobile% ls
babel-plugin-transform-runtime/  codeframe/   fs-search/               markdown-ansi/      packager-raw/           transformer-inline-string/  transformer-react-refresh-babel/
babel-preset-env/                diagnostic/  fs-write-stream-atomic/  namer-default/      plugin/                 transformer-json/           types/
babylon-walk/                    events/      logger/                  node-libs-browser/  runtime-react-refresh/  transformer-raw/            watcher/
Desktop-cf97a:/private/var/tmp/parcel-doesnt-declare-its-variables/node_modules/@parcel mobile%
```

How can I prevent that/manually download the files?

**Reply by mischnic** [reply]
Running this script results in the syntax error: https://unpkg.com/browse/node-gyp-build@4.2.3/bin.js

So just try running `npx node-gyp-build` anywhere. Looks like the shebang to run this file via Node is ignored on your device?

**Reply by danieltroger** [reply]
> Looks like the shebang to run this file via Node is ignored on your device?

Oh my god, you're a genius. I added this on the second line `node <<EOF` and this on the last line `EOF` and now I'm getting this error:

```
internal/child_process.js:402
    throw errnoException(err, 'spawn');
    ^

Error: spawn EPERM
    at ChildProcess.spawn (internal/child_process.js:402:11)
    at Object.spawn (child_process.js:553:9)
    at build ([stdin]:27:8)
    at preinstall ([stdin]:36:32)
    at [stdin]:9:7
    at ChildProcess.exithandler (child_process.js:315:5)
    at ChildProcess.emit (events.js:314:20)
    at maybeClose (internal/child_process.js:1047:16)
    at Socket.<anonymous> (internal/child_process.js:438:11)
    at Socket.emit (events.js:314:20) {
  errno: -1,
  code: 'EPERM',
  syscall: 'spawn'
}
```

Does that mean iOS's `fork()` protection is somehow not properly patched away?
That would be sad af 😭 Do you happen to know a workaround for that? 🙃

(@Diatrus I think you compiled the node for iOS that I'm using, have you ran into this error before?)

**Reply by Diatrus** [reply]
@danieltroger indeed I have... I can have a look within the day. Could you make an issue on the Procursus github? (EPERM error NodeJS)

**Reply by danieltroger** [reply]
Thanks @Diatrus. I opened https://github.com/ProcursusTeam/Procursus/issues/689. Let me know if you have trouble reproducing.
